File: 1341-split-a-string-in-balanced-strings/split-a-string-in-balanced-strings.py
class Solution(object):
    def balancedStringSplit(self, s):
        """
        :type s: str
        :rtype: int
        """
        c=0
        l=0
        r=0
        for x in s:
            if x=='L': l+=1
            if x=='R': r+=1
            if l==r: c+=1
        return c

        

File: 1341-split-a-string-in-balanced-strings/split-a-string-in-balanced-strings.java
class Solution {
    public int balancedStringSplit(String s) {
        int n=0;
        int r=0,l=0;
        for(char c:s.toCharArray()){
            if (c=='R')
            r++;
            if(c=='L')
            l++;
            if(r==l) n++;
        }
        return n;
    }
}